org.mwc.cmap.NarrativeViewer/src/org/mwc/cmap/NarrativeViewer/NarrativeViewer.java

```diff
@@ -249,7 +250,30 @@ public void setDTG(final HiResDate dtg)
     // ok, try to select this entry
     if (entry != null)
     {

```

Comment:
@theanuradha - I introduced this extra step to only change the selected item in the viewer if it's different to the one that's currently selected.

Does it look ok?